clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -63,196 +87,59 @@ public void testSerdeFromNotNull() {
-
-        for (Integer integer : integers) {
-            assertEquals("Should get the original integer after serialization and deserialization",
-                    integer, deserializer.deserialize(topic, serializer.serialize(topic, integer)));
-        }
-
-        assertEquals("Should support null in serialization and deserialization",
-                null, deserializer.deserialize(topic, serializer.serialize(topic, null)));
-
-        serializer.close();
-        deserializer.close();
-    }
-
-    @Test
-    public void testLongSerializer() {
-        Long[] longs = new Long[]{
-            922337203685477580L,
-            -922337203685477581L
-        };
-
-        Serializer<Long> serializer = Serdes.Long().serializer();
-        Deserializer<Long> deserializer = Serdes.Long().deserializer();
-
-        for (Long value : longs) {
-            assertEquals("Should get the original long after serialization and deserialization",
-                    value, deserializer.deserialize(topic, serializer.serialize(topic, value)));
-        }
-
-        assertEquals("Should support null in serialization and deserialization",
-                null, deserializer.deserialize(topic, serializer.serialize(topic, null)));
-
-        serializer.close();
-        deserializer.close();
-    }
+            try (Serde<String> serDeser = getStringSerde(encoding)) {
 
-    @Test
-    public void shouldSerializeDeserializeFloat() {
-        final Float[] floats = new Float[]{
-            5678567.12312f,
-            -5678567.12341f
-        };
-        final Serializer<Float> serializer = Serdes.Float().serializer();
-        final Deserializer<Float> deserializer = Serdes.Float().deserializer();
-
-        for (final Float value : floats) {
-            assertThat("Should round-trip a float",
-                value, equalTo(deserializer.deserialize(topic, serializer.serialize(topic, value))));
+                Serializer<String> serializer = serDeser.serializer();
+                Deserializer<String> deserializer = serDeser.deserializer();
+                assertEquals("Should get the original string after serialization and deserialization with encoding " + encoding,
+                        str, deserializer.deserialize(topic, serializer.serialize(topic, str)));
+            }
         }
-
-        serializer.close();
-        deserializer.close();
-    }
-
-    @Test
-    public void floatSerializerShouldReturnNullForNull() {
-        final Serializer<Float> serializer = Serdes.Float().serializer();
-        assertThat(serializer.serialize(topic, null), nullValue());
-        serializer.close();
-    }
-
-    @Test
-    public void floatDeserializerShouldReturnNullForNull() {
-        final Deserializer<Float> deserializer = Serdes.Float().deserializer();
-        assertThat(deserializer.deserialize(topic, null), nullValue());
-        deserializer.close();
     }
 
-    @Test
+    @Test(expected = SerializationException.class)
```

Comment:
IIRC we do not want to use the `@Test(expected = ...)` pattern here, /cc @dguy.

Also, we are not calling `close()` any longer, which IMHO we should because of the general serializer/deserializer contract.